br/pkg/lightning/mydump/region.go

```diff
@@ -310,13 +311,18 @@ func makeParquetFileRegion(
 	dataFile FileInfo,
 	prevRowIdxMax int64,
 ) (int64, *TableRegion, error) {
-	r, err := store.Open(ctx, dataFile.FileMeta.Path)
-	if err != nil {
-		return prevRowIdxMax, nil, errors.Trace(err)
-	}
-	numberRows, err := ReadParquetFileRowCount(ctx, store, r, dataFile.FileMeta.Path)
-	if err != nil {
-		return 0, nil, errors.Trace(err)
+	var numberRows int64 = 0
+
+	// for S3, reading an empty file will cause error, so skip reading the empty files
```

Comment:
Nit: An alternative way is to fix the issue that we can't reading an empty file from S3. In external storage, if the given file size is zero, just return an `eofReadCloser`.

Reply:
I think return an `eofReadCloser` is not ok for some cases. Especially, when we treat a 'directory' as a parquet file, because an 'empty' parquet file still contains head and footer, it will cause open the parquet file fails.

Reply:
Yes. This can be two different issues:

1. Skip fake directory file in S3.
2. Any implementation of external storage shouldn't return error when file is empty. 

This PR can only process issue 1.

Reply:
> Nit: An alternative way is to fix the issue that we can't reading an empty file from S3. In external storage, if the given file size is zero, just return an `eofReadCloser`.

Currently it is hard to do that because the input parameters of `ExternalStorage.Open()` has no size information.

Reply:
You can use `S3API.HeadObject` to get the size information.